pkg/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -0,0 +1,557 @@
+package regopolicyinterpreter
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage/inmem"
+	"github.com/open-policy-agent/opa/topdown"
+	"github.com/pkg/errors"
+)
+
+type LogLevel int
+
+const (
+	LogNone LogLevel = iota
+	// Logs the output of Rego print() statements in the policy
+	LogInfo
+	// Logs the result objects returned from each query
+	LogResults
+	// Logs the full metadata state after each query
+	LogMetadata
+)
+
+type RegoPolicyInterpreter struct {
+	// Mutex to prevent concurrent access to fields
+	mutex sync.Mutex
+	// Rego which describes policy behavior (see above)
+	code string
+	// Rego data namespace
+	data map[string]interface{}
+	// Modules
+	modules map[string]*RegoModule
+	// Compiled modules
+	compiledModules *ast.Compiler
+	// Logging
+	logLevel       LogLevel
+	logFile        *os.File
```

Comment:
consider integrating with logrus instead. Ideally, we'd use the context we create during processing bridge requests, that way we'll be able to go back to the UVM/container IDs, but this will involve refactoring how the security policy enforcer interface looks like and out of scope of this PR.

Reply:
I'll add this as a Nit for now. I think there is an argument for leaving it with its own logging system (e.g. if it needed to be used in other places), not to mention that it almost certainly should not be turned on except when the package is being used by something like the simulator, which uses file-based logs and not event-based ones for ease of use for the tester.